crates/ignore/src/dir.rs

```diff
@@ -507,6 +516,86 @@ impl Ignore {
     fn absolute_base(&self) -> Option<&Path> {
         self.0.absolute_base.as_ref().map(|p| &***p)
     }
+
+    pub fn get_match_metadata<'self_>(
+        &'self_ self,
+        match_metadata_token: MatchMetadataToken,
+    ) -> MatchMetadata<'self_> {
+        MatchMetadata {
+            matching_file_types: MatchingFileTypesIterator::new(
+                self,
+                match_metadata_token,
+            ),
+        }
+    }
+
+    pub fn get_file_type_def_at_selection_index(
+        &self,
+        selection_index: usize,
+    ) -> &FileTypeDef {
+        self.0.types.get_file_type_def_at_selection_index(selection_index)
+    }
+
+    pub fn should_skip_entry_with_match_metadata_token(
```

Comment:
This basically got refactored into a method here (from a top-level function helper `should_skip_entry()`)